### ISSUE TITLE
Is there any way to re-encrypt an ebook with DRM using this library/code?

### ISSUE THREAD

**Issue by seidnerj** [noDRM/DeDRM_tools#605]
### Question / bug report

Is this even be possible?

### Which version of Calibre are you running?

N/A

### Which version of the DeDRM plugin are you running?

N/A

### Log output

```log
Paste log output here.
```

**Comment by ElleKayEm** [comment]
These tools unencrypt specific types of DRM.  The encryption is Amazon's, Kobo's, Adobe's, etc.

**Comment by seidnerj** [comment]
Yes. I'll clarify - is there a way to add Amazon's DRM to an ebook using this (or some other) library?

**Comment by ElleKayEm** [comment]
Publishing through Amazon.

**Comment by j-howell** [comment]
I don’t know of anything that can do that and I am curious why you would want to.

Are you trying to produce DRM protected books to sell to Kindle users without going through Amazon? Are you trying to re-encrypt a book that you have decrypted and then modified in some way? Or something else?

**Comment by seidnerj** [comment]
> Publishing through Amazon.

Yes, that is obvious. I am asking if there's another way to do this.

**Comment by seidnerj** [comment]
> I don’t know of anything that can do that and I am curious why you would want to.
> 
> Are you trying to produce DRM protected books to sell to Kindle users without going through Amazon? Are you trying to re-encrypt a book that you have decrypted and then modified in some way? Or something else?

The latter, I am looking to re-encrypt a book that has been decrypted & altered, in order to make sure the DRM still protects the altered version of the content.

**Comment by j-howell** [comment]
Ok. I don't know who you would be protecting it from since any DRM you can remove someone else can remove also.

In any case, it should be possible to reverse the DRM removal algorithm but I don't know of anyone actually doing that. I doubt that there is much interest in such a capability.

**Comment by seidnerj** [comment]
The point is a legal one, to not provide non-DRM protected copies to end users.

**Comment by ElleKayEm** [comment]
Ok, still baffled.  Sounds like you are trying to do exactly the sort of thing that DRM is intended to protect against unless you are the copyright holder or have their permission.  In which case, not sure why you would need to DIY DRM.

**Comment by seidnerj** [comment]
Well, yes and no. DRM's main purpose is to protect against content "theft", which is the context of eBooks pertains mostly do creating copies of the content and sending it to parties that have not purchased the content.

Depending on the geography, it may or may not be legal to remove DRM (for personal use, if you own the content, or even in general). The party that's mostly exposed legally, if set up correctly, is the end user. If you could provide the user with a similarly protected eBook, locked to their original device this would go a long ways to protecting both the end user and the 3rd party manipulating the content on their behalf.

**Comment by ElleKayEm** [comment]
Tying it back to Amazon's DRM seems like a spectacularly bad idea to me, and I don't agree that it would protect either end user or third party.  But I'm not interested in arguing about it so I'll just say thanks for your explanation and leave it at that.

**Comment by seidnerj** [comment]
> Tying it back to Amazon's DRM seems like a spectacularly bad idea to me, and I don't agree that it would protect either end user or third party.  But I'm not interested in arguing about it so I'll just say thanks for your explanation and leave it at that.

Oh! so this how someone who is not interested in arguing usually responds. I almost did not get that.

**Comment by ElleKayEm** [comment]
Like j-howell, I was curious why.  Now my curiosity is satisfied.
